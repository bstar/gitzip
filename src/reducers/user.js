import {
    SET_GITHUB_KEY,
    GET_USER_SUCCESS,
    GET_USER_REPOS_SUCCESS,
    SET_ACTIVE_REPO,
    GET_USER_REPO_ISSUES_SUCCESS,
} from '../constants/ActionTypes';

const initialSettings = {};

const user = (state = initialSettings, action) => {

    switch (action.type) {

        case SET_GITHUB_KEY:

            return {
                ...state,
                key: action.payload,
            };

        case GET_USER_SUCCESS:

            return {
                ...state,
                login: action.payload.login,
                avatar_url: action.payload.avatar_url,
                url: action.payload.url,
                name: action.payload.name,
            };

        case GET_USER_REPOS_SUCCESS:

            return {
                ...state,
                repos: action.payload,
            };

        case SET_ACTIVE_REPO:

            return {
                ...state,
                activeRepoData: action.payload,
                issues: null,
            };

        case GET_USER_REPO_ISSUES_SUCCESS:

            return {
                ...state,
                issues: action.payload,
            };

        default:
            return state;
    }
};

export default user;
