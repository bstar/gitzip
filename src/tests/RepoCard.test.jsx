import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import reducers from '../reducers';
import RepoCard from '../components/repoCard';


const store = createStore(
    reducers,
);

it('renders correctly', () => {
    const tree = renderer.create(<Provider store={store}><RepoCard repo={{ name: 'test' }} /></Provider>).toJSON();

    expect(tree).toMatchSnapshot();
});
