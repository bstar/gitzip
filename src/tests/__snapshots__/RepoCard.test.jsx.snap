// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="repo-card-container"
  onClick={[Function]}
  style={Object {}}
>
  <div>
    <b>
      Name: 
      test
    </b>
  </div>
  <div>
    <b>
      Description:
    </b>
     
  </div>
</button>
`;
